### ISSUE TITLE
Symbol is not displayed  in legend in Mapserver 7 without keyimage

### ISSUE THREAD

**Issue by wirkus** [mapserver/mapserver#5096]
Hello,

I'm testing Mapserver7. Everything is OK, but I have problems with symbols in the legend.
The mapserver-symbols and png-symnbols are displayed in the Map but not in the legend, see below.
IF I use a KEYIMAGE, the legend is displayed.

SIZEUNITS meters
CLASS
NAME "Eingriff"
STYLE
SYMBOL 'circle'
COLOR 255 20 0
SIZE 12
MINSIZE 8
MAXSIZE 16
OUTLINECOLOR 0 0 0
WIDTH 1
MINWIDTH 1
MAXWIDTH 1
END
LABEL
TYPE TRUETYPE
#COLOR 0 255 55
COLOR 255 0 0
OUTLINECOLOR 254 254 254
FONT "Arial"
POSITION uc
PARTIALS FALSE
FORCE TRUE
MINDISTANCE -1
SIZE 9
MINSIZE 8
MAXSIZE 12
END
END

Do I have to define a keyimage?

Best regards

**Comment by tbonfort** [comment]
I suspect this has something to do with using SIZEUNITS. What happens if you remove that?

**Comment by astroidex** [comment]
Yes, @tbonfort. It has to do with using SIZEUNITS. @Schpidi and I were testing the issue at FOSS4G code sprint. @Schpidi changed the mapfile https://github.com/mapserver/msautotest/blob/master/renderers/legend.map in every layer we said SIZEUNITS meters. The effect was:

* labeling in the legend is much smaller -> depends on the SIZE of label in CLASS
* some symbols are not displayed at all (all point symols)
* polygon symbols - outline is not displayed at all

**Comment by Schpidi** [comment]
Original:
![legend](https://cloud.githubusercontent.com/assets/482265/8761506/f8c80b5c-2d54-11e5-90d7-fa8bd023a812.png)

SIZEUNTES METERS:
![legend](https://cloud.githubusercontent.com/assets/482265/8761507/fe1c46b8-2d54-11e5-9907-4b1e80de5471.png)

**Comment by wirkus** [comment]
Hello,
we tested point symbols in the legend again:
 * SIZEUNITS pixel is OK
 * SIZEUNITS meters: umn symbol: no image in the legend but text
 * SIZEUNITS meters: png symbol: no image in the legend but text
 * SIZEUNITS meters: ttf-Symbol: no response from mapserver; mapserver blocks more or less the whole cpu

Best regards

**Comment by tbonfort** [comment]
Outdated issue after #5268. Please open a new issue with a testcase if this one is still relevant.
